src/main/java/commands/PlanCommand.java

```diff
@@ -1,11 +1,145 @@
 package commands;
 
+import data.plans.InvalidPlanException;
+import data.plans.Plan;
+import data.plans.PlanList;
+import storage.FileManager;
+import werkit.UI;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 public class PlanCommand extends Command {
-    public PlanCommand(String userInput) {
+    public static final String BASE_KEYWORD = "plan";
+    public static final String CREATE_ACTION_KEYWORD = "/new";
+    public static final String CREATE_ACTION_WORKOUTS_KEYWORD = "/workouts";
+    public static final String LIST_ACTION_KEYWORD = "/list";
+    public static final String DETAILS_ACTION_KEYWORD = "/details";
+    public static final String DELETE_ACTION_KEYWORD = "/delete";
+
+    private FileManager fileManager;
+    private UI ui = new UI();
+    private PlanList planList;
+
+    private String userAction;
+    private String userArguments;
+
+    private static Logger logger = Logger.getLogger(WorkoutCommand.class.getName());
+
+    public PlanCommand(String userInput, FileManager fileManager, PlanList planList,
+                       String userAction, String userArguments) throws InvalidCommandException {
         super(userInput);
+        this.fileManager = fileManager;
+        this.planList = planList;
+        setUserAction(userAction);
+        this.userArguments = userArguments;
     }
```

Comment:
Should include `LogHandler.linkToFileLogger(logger);` to get the log messages in this class to be written to the log file.

Reply:
added!